javatests/arcs/core/util/ParserTest.kt

```diff
@@ -0,0 +1,157 @@
+package arcs.core.util
+
+import arcs.core.util.ParseResult.Failure
+import arcs.core.util.ParseResult.Success
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import kotlin.test.assertFails
+
+
+/** Tests for [Parser]. */
+@RunWith(JUnit4::class)
+class ParserTest {
+
+    fun String.rest(pos: SourcePosition) =
+        if (pos.offset >= this.length) "" else this.substring(pos.offset)
+
+    @Test
+    fun parseWithTokenizer() {
+        val hello = token("hello")
+        hello("hello world", 0).map { match, start, end ->
+            assertThat(start).isEqualTo(SourcePosition(0, 0, 0))
```

Comment:
if `.map {...` does not execute in case of parse failure, do you have guarantee that these assertions actually run?

Reply:
added orElse<Nothing> { throw ... }